worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgePlayer.java

```diff
@@ -183,18 +188,34 @@ public boolean hasPermission(String perm) {
 
     @Override
     public <B extends BlockStateHolder<B>> void sendFakeBlock(BlockVector3 pos, B block) {
+        World world = getWorld();
+        if (!(world instanceof ForgeWorld)) {
+            return;
+        }
         BlockPos loc = ForgeAdapter.toBlockPos(pos);
         if (block == null) {
-            // TODO
-//            player.sendBlockChange(loc, player.getWorld().getBlockAt(loc).getBlockData());
+            final SChangeBlockPacket packetOut = new SChangeBlockPacket(((ForgeWorld) world).getWorld(), loc);
+            player.connection.sendPacket(packetOut);
         } else {
-            // TODO
-//            player.sendBlockChange(loc, BukkitAdapter.adapt(block));
-            if (block instanceof BaseBlock && ((BaseBlock) block).hasNbtData()) {
-                player.connection.sendPacket(new SPacketUpdateTileEntity(
-                        new BlockPos(pos.getBlockX(), pos.getBlockY(), pos.getBlockZ()), 7,
-                        NBTConverter.toNative(((BaseBlock) block).getNbtData()))
-                );
+            final SChangeBlockPacket packetOut = new SChangeBlockPacket();
+            PacketBuffer buf = new PacketBuffer(Unpooled.buffer());
+            buf.writeBlockPos(loc);
+            buf.writeVarInt(Block.getStateId(ForgeAdapter.adapt(block.toImmutableState())));
+            try {
+                packetOut.readPacketData(buf);
+            } catch (IOException e) {
+                return;
+            }
+            player.connection.sendPacket(packetOut);
+            if (block instanceof BaseBlock) {
+                final BaseBlock baseBlock = (BaseBlock) block;
+                final CompoundTag nbtData = baseBlock.getNbtData();
+                if (nbtData != null) {
+                    player.connection.sendPacket(new SUpdateTileEntityPacket(
+                            new BlockPos(pos.getBlockX(), pos.getBlockY(), pos.getBlockZ()), 7,
```

Comment:
Can we put that `7` in a constant field, named `STRUCTURE_BLOCK_TILE_ENTITY_ID` or something? Honestly, we should be careful with this, because `7` won't work if we start doing non-structure blocks. Perhaps we should have a map of IDs?